Homeworks/Hw9/hw.py

```diff
@@ -0,0 +1,119 @@
+import pytest
+import time
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support import expected_conditions as EC
+from selenium.common.exceptions import StaleElementReferenceException
+from selenium.webdriver.support.wait import WebDriverWait
+
+
+
+def test_login(driver):
+    driver.get("https://www.demoblaze.com/")
+
+
+    login_button = WebDriverWait(driver, 10).until(
+        EC.element_to_be_clickable((By.ID, "login2"))
+    )
+    login_button.click()
+
+    username_field = WebDriverWait(driver, 10).until(
+        EC.element_to_be_clickable((By.ID, "loginusername"))
+    )
+    username_field.send_keys("danil666")
+
+    password_field = driver.find_element(By.ID, "loginpassword")
+    password_field.send_keys("Qwerty123")
+
+    login_button1 = driver.find_element(By.XPATH, "//button[contains(text(), 'Log in')]")
+    login_button1.click()
+
+    time.sleep(5)
+
+    logout_button = driver.find_element(By.ID, "logout2")
+
+    assert logout_button.is_displayed()
+    assert "Welcome danil666" in driver.find_element(By.XPATH, "/html/body").text
+
+def test_add_to_cart(driver):
+    driver.get("https://www.demoblaze.com/")
+
+
+    wait = WebDriverWait(driver, timeout=5)
+
+    login_button = WebDriverWait(driver, 5).until(
+        EC.element_to_be_clickable((By.ID, "login2"))
+    )
+    login_button.click()
+
+    username_field = WebDriverWait(driver, 5).until(
+        EC.element_to_be_clickable((By.ID, "loginusername"))
+    )
+    username_field.send_keys("danil666")
+
+    password_field = driver.find_element(By.ID, "loginpassword")
+    password_field.send_keys("Qwerty123")
+
+    login_button1 = driver.find_element(By.XPATH, "//button[contains(text(), 'Log in')]")
+    login_button1.click()
+
+    monitors_element_found = False
+    while not monitors_element_found:
+        try:
+            monitors_element = WebDriverWait(driver, 20).until(
+                EC.element_to_be_clickable((By.XPATH, "//a[@onclick=\"byCat('monitor')\"]"))
+            )
+            monitors_element.click()
+            monitors_element_found = True
+        except StaleElementReferenceException:
+            pass
+
+    wait.until(EC.presence_of_all_elements_located((By.XPATH, "//div[@class='card h-100']")))
+
+    products = driver.find_elements(By.XPATH, "//div[@class='card h-100']")
+
+    max_price = 0
+    max_price_elem = None
+    for product in products:
```

Comment:
it would be better maintain tests readable and simple and use helpers modules for complex logic
also using page object model is a best practice